_source/_posts/2023-07-28-scim-workshop.md

```diff
@@ -0,0 +1,1144 @@
+
+On the command line, `npm install passport-http-bearer` to add the library to your project.
+
+In `scim.ts`, import `body-parser`:
+```
+import bodyParser from 'body-parser';
+```
+And instruct the app to use `bodyParser` for the relevant content types: 
+
+```
+app.use(bodyParser.json({ type: ['application/scim+json'] }));
+```
+Sign up for [Postman](https://identity.getpostman.com/login) or sign in to your account, and configure it to communicate with your local instance of the Todo app. 
+
+In Postman, the request URL will be`http://localhost:3333/scim/v2/Users` if you're running the Todo app locally. In the Headers tab, add the key `Content-Type` and set its value to `application/scim+json`, and then add an additional key, `Authorization`, and set it to `Bearer 131313`.This bearer token value comes from the `apikey` variable set earlier in `seed.ts`. 
+
+
+Now we are ready to test with Postman with our local server. You can also make cURL requests directly on the terminal if you prefer. 
+
+##### Try a  POST request 
+Try sending the following request to `http://localhost:3333/scim/v2/Users`:
+
+```
+{
+    "schemas": [
+        "urn:ietf:params:scim:schemas:core:2.0:User"
+    ],
+    "userName": "test.user@portal.example",
+    "name": {
+        "givenName": "Test",
+        "familyName": "User"
+    },
+    "emails": [
+        {
+            "primary": false,
+            "value": "test.user@portal.example",
+            "type": "work"
+        },
+        {
+            "primary": true,
+            "value": "test.user@portal.example",
+            "type": "work"
+        }
+    ],
+    "displayName": "Test User",
+    "locale": "en-US",
+    "externalId": "00ujl29u0le5T6Aj10h7",
+    "groups": [],
+    "password": "1mz050nq",
+    "active": true
+}
+```
+
+What response do you expect? If your code is working correctly, the Todo app backend will give a response with status 201 Created, and the following body: 
+
+```
+{
+    "schemas": [
+        "urn:ietf:params:scim:schemas:core:2.0:User"
+    ],
+    "locale": "en-US",
+    "groups": [],
+    "meta": {
+        "resourceType": "User"
+    },
+    "id": "5",
+    "userName": "test.user@portal.example",
+    "name": {
+        "givenName": "Test",
+        "familyName": "User"
+    },
+    "emails": [
+        {
+            "primary": true,
+            "value": "test.user@portal.example",
+            "type": "work"
+        }
+    ],
+    "displayName": "Test User",
+    "externalId": "00ujl29u0le5T6Aj10h7",
+    "active": true
+}
+```
+
+If you send the same POST request again, re-creating an existing user, what should happen? Try it, and you'll get a 409 Conflict response, with a body like this: 
+
+```
+{
+    "schemas": [
+        "urn:ietf:params:scim:api:messages:2.0:Error"
+    ],
+    "detail": "User already exists in the database: 5",
+    "status": 409
+}
+```
+
+You can repeat this testing process for each SCIM route that you implement! 
+
+### GET /Users
```

Comment:
This needs an explanation of why Okta will be making a GET request to list users in your database.